network/horizon/admin-guide/error-handling.mdx

```diff
@@ -0,0 +1,182 @@
+
+#### Async Transaction Submission
+
+Submissions using the async endpoint return a clear response back from core. There are different actions that clients can take based on the specific `tx_status` returned:
+
+1. `PENDING`: the submission is successful but the txn is still waiting to be included in the ledger. You should use a `GET` endpoint like `/transactions` to poll for the submitted txn and check if it makes into the ledger.
+2. `ERROR`: the submission did not go through due to some error with core. Take a look at the attached error message and retry the submission again.
+3. `TRY_AGAIN_LATER`: this indicates that the stellar-core instance is currently unable to process any submission requests. Clients should wait for sometime before retrying it again.
+
+#### Non-Async Transaction Submission
+
+Due to the blocking nature of the non-async endpoint, things are a little different compared to the async strategy. There are 3 possible scenarios that clients can encounter:
+
+1. The submission is successful and Horizon returns the transaction response back. This is the happy path and clients do not need to do anything but wait for Horizon's response.
+2. Core sends back an `ERROR` response from the submission. Clients should consult the attached error message and retry the submission again.
+3. Timeouts: Horizon may send a `504 Timeout` after transaction submission. Timeouts are not errors but warnings that your request hasn’t been fulfilled yet. This can happen because of the relationship between Horizon and Stellar Core- the network may take some time (5-10 mins during congestion) to accept the transaction. At the same time, Horizon needs to provide developers with a response within 30 seconds. Due to the blocking nature of the endpoint, there could be many possible scenarios behind the timeout - transient outages, congestion on the network, issues on Core side, issues on Horizon side etc.... There is no way to cancel a transaction after submitting it. So, you need to carefully consider before resubmitting your transactions.
+   1. `Polling the transaction hash`: The recommended step is to use the txn hash in the timeout response and poll the `GET /transactions` endpoint to see if it successfully made it into a ledger.
+   2. `Time bounds`: Time bounds are optional but recommended, as they put a time limit on the transaction - so either the transaction makes it onto the ledger or it times out depending on your time parameters.
+   3. `Resubmitting the transaction`: If the txn status poll has not returned anything or if your transaction has timed out based on the time bounds you specified, you should attempt a resubmission of the same transaction. If the transaction has already successfully made it to the ledger, Horizon will return any attempted resubmission. Only in cases where a transaction’s status is unknown (and thus will have a chance to make it on the ledger) will a resubmission to the network occur.
```

Comment:
I don't think we want to mention this, as it contracts the suggestion to poll using the transaction hash.

Reply:
Agreed, we should discourage relying on Horizon's unnecessary helpfulness in this case